src/featurestorebundle/delta/feature/NullHandler.py

```diff
@@ -16,9 +20,11 @@ def fill_nulls(self, df: DataFrame, feature_list: FeatureList) -> DataFrame:
 
         return df.fillna(fill_dict)
 
-    def to_storage_format(self, df: DataFrame, feature_list: FeatureList) -> DataFrame:
+    def to_storage_format(self, df: DataFrame, feature_list: FeatureList, entity: Entity) -> DataFrame:
+        self.__features_validator.validate(entity, df, feature_list)
```

Comment:
can it even happen that the metadata don't fit data during write?

Reply:
I don't think so but better be safe than sorry